lib/widgets/app.dart

```diff
@@ -152,18 +152,56 @@ class _ZulipAppState extends State<ZulipApp> with WidgetsBindingObserver {
     return super.didPushRouteInformation(routeInformation);
   }
 
-  Future<void> _handleInitialRoute() async {
-    final initialRouteUrl = Uri.parse(WidgetsBinding.instance.platformDispatcher.defaultRouteName);
-    if (initialRouteUrl case Uri(scheme: 'zulip', host: 'notification')) {
-      await NotificationDisplayManager.navigateForNotification(initialRouteUrl);
+  InitialRouteListFactory _handleGenerateInitialRoutes(BuildContext context) {
+    void showNotificationErrorDialog() async {
+      final navigator = await ZulipApp.navigator;
+      final navigatorContext = navigator.context;
+      assert(navigatorContext.mounted);
+      // TODO(linter): this is impossible as there's no actual async gap, but
+      //   the use_build_context_synchronously lint doesn't see that.
+      if (!navigatorContext.mounted) return;
+
+      final zulipLocalizations = ZulipLocalizations.of(navigatorContext);
+      showErrorDialog(context: navigatorContext,
+        title: zulipLocalizations.errorNotificationOpenTitle,
+        message: zulipLocalizations.errorNotificationOpenAccountMissing);
     }
+
+    final globalStore = GlobalStoreWidget.of(context);
+    // TODO(#524) choose initial account as last one used
+    final initialAccountId = globalStore.accounts.firstOrNull?.id;
```

Comment:
How about doing this (checking the contents of `globalStore.accounts`) inside the returned `InitialRouteListFactory` callback? That's what we do in the open-from-notification case.

Then in particular we don't have to think about what happens if the `firstOrNull` account is logged out between a call to `_ZulipAppState` and the time when Flutter decides to call this `InitialRouteListFactory` callback.

—ah, I see: its position outside the callback is the same in `main`. How about a prep commit that just puts the query on `globalStore.accounts` into the callback, then?

Reply:
Seems also nice if it's below the open-from-notification code in that callback, so it's easier to see that it's not part of that logic.